internal/service/ec2/transitgateway_shared_owners.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceTransitGatewaySharedOwners() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: ResourceTransitGatewaySharedOwnersCreate,
+		ReadWithoutTimeout:   ResourceTransitGatewaySharedOwnersRead,
+		DeleteWithoutTimeout: ResourceTransitGatewaySharedOwnersDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTransitGatewaySharedOwnersImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"transit_gateway_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func ResourceTransitGatewaySharedOwnersCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).EC2Conn
+
+	transitGatewayID := d.Get("transit_gateway_id").(string)
+	accountID := d.Get("account_id").(string)
+
+	transitGateway, err := DescribeTransitGateway(conn, transitGatewayID)
+	if err != nil {
+		return diag.Errorf("error describing Transit Gateway (%s): %s", transitGatewayID, err)
+	}
+
+	if transitGateway == nil {
+		return diag.Errorf("Transit Gateway %s not found", transitGatewayID)
+	}
+
+	modifyInput := &ec2.ModifyTransitGatewayInput{
+		TransitGatewayId: aws.String(transitGatewayID),
+		Options: &ec2.ModifyTransitGatewayOptions{
+			AddSharedOwners: []*string{aws.String(accountID)},
+		},
+	}
+
+	_, err = conn.ModifyTransitGateway(modifyInput)
+	if err != nil {
+		return diag.Errorf("[ERROR] Error modifying Transit Gateway (%s) to add shared owner (%s): %s", transitGatewayID, accountID, err)
+	}
+
+	id := TransitGatewaySharedOwnersCreateResourceID(transitGatewayID, accountID)
+	d.SetId(id)
+
+	return ResourceTransitGatewaySharedOwnersRead(ctx, d, meta)
+}
+
+func ResourceTransitGatewaySharedOwnersRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).EC2Conn
+
+	transitGatewayID := d.Get("transit_gateway_id").(string)
+	transitGateway, err := DescribeTransitGateway(conn, transitGatewayID)
+
+	if err != nil {
+		return diag.Errorf("reading Transit Gateway (%s): %s", transitGatewayID, err)
+	}
+
+	if transitGateway == nil || transitGateway.Options == nil {
+		log.Printf("[WARN] Transit Gateway %s not found, removing from state", transitGatewayID)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("transit_gateway_id", transitGatewayID)
+
+	accountID := d.Get("account_id").(string)
+	d.Set("account_id", accountID)
+
+	if !d.IsNewResource() {
+		if transitGateway.Options != nil && transitGateway.Options.SharedOwners != nil {
+			AccountIDExist := false
+			for _, owner := range transitGateway.Options.SharedOwners {
+				if owner != nil && aws.StringValue(owner) == accountID {
```

Comment:
```suggestion
				if aws.StringValue(owner) == accountID {
```
Если в `accountID` точно не пустая строка, то этого достаточно, тк `aws.StringValue` конвертирует в том числе `nil`